Solaris 9 struct addrinfo had incorrect size of member ai_addrlen. Details are in
https://bugs.squid-cache.org/show_bug.cgi?id=3057

The issue is fixed in Solaris 10. Instead now it creates problems for sparc
builds on Solaris 11. The fix is to remove the no longer needed workaround.

--- squid-6.9/compat/os/solaris.h
+++ squid-6.9/compat/os/solaris.h
@@ -81,9 +81,11 @@ SQUIDCEXTERN int gethostname(char *, int
  *           this causes binary incompatibility on 64-bit systems.
  *  Fix this by bundling a copy of the OpenSolaris 10 netdb.h to use instead.
  */
+/*
 #if defined(__sparcv9)
 #include "compat/os/opensolaris_10_netdb.h"
 #endif
+*/
 
 /* Solaris 10 lacks SUN_LEN */
 #if !defined(SUN_LEN)
